File: 02-moderate/point_in_circle/point.in.circle.py
# Reuben Thorpe (2016), CodeEval [Point In Circle v1.0]
from sys import argv
from math import sqrt
import re


def parse_problem(file_path):
    float_regex = re.compile(r"[-+]?\d*\.\d+|[-+]?\d+")

    with open(file_path, "r") as in_file:
        for line in open(argv[1], "r"):
            parse_floats = [float(num) for num in re.findall(float_regex, line)]

            yield({"center": parse_floats[0:2],
                   "radius": parse_floats[2],
                   "point": parse_floats[3:5]})


def distance(x, y):
    """
        Return the distance between two points.
    """

    x1, y1 = x
    x2, y2 = y

    return(sqrt(pow(x2 - x1, 2) + pow(y2 - y1, 2)))


def is_in_circle(problem):
    """
        Test whether a point is lying within a circle.
    """
    return(distance(problem["center"], problem["point"]) < problem["radius"])


if __name__ == "__main__":
    for problem in parse_problem(argv[1]):
        print("true") if is_in_circle(problem) else print("false")

File: code.eval/02-moderate/point_in_circle/point.in.circle.cpp
// Reuben Thorpe (2016), CodeEval [Point In Circle v1.1]
#include <iostream>
#include <fstream>
#include <stdio.h>
#include <math.h>

using namespace std;

float distance(float x1, float y1, float x2, float y2);


int main(int argc, char* argv[]) {
  ifstream in_file(argv[1]);
  string line;
  float center_x, center_y, radius, point_x, point_y;
  cout << boolalpha;

  while(getline(in_file, line)) {

    sscanf(line.c_str(),
           "Center: (%f, %f); Radius: %f; Point: (%f, %f)",
           &center_x, &center_y, &radius, &point_x, &point_y);

    cout << (distance(center_x, center_y, point_x, point_y) < radius) << '\n';
  }

  return(0);
}


float distance(float x1, float y1, float x2, float y2) {
  // Measure the distance between two points.
  return(sqrt(pow(x2 - x1, 2) + pow(y2 - y1, 2)));
}
